pkg/controller/provider/controller.go

```diff
@@ -196,11 +198,46 @@ func (r Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (r
 			deployment)
 
 		// If the deployment does not exist
-		if k8serr.IsNotFound(err) {
-			r.CreateOVAServerDeployment(provider, ctx)
-		} else if err != nil {
+		if err != nil {
+			if k8serr.IsNotFound(err) {
+				r.CreateOVAServerDeployment(provider, ctx)
+				provider.Status.Phase = initializing
+				provider.Status.SetCondition(
+					libcnd.Condition{
+						Type:     initializing,
+						Status:   True,
+						Category: Required,
+						Message:  "The OVA server being inizialized.",
+					})
+				err = r.Status().Update(context.TODO(), provider.DeepCopy())
```

Comment:
The `.DeepCopy()` func was missing for copying the correct values.

Reply:
We talked about it in the sync, the status was applied and everything was reset after it, so I just added a comment that instead using simply `err = r.Status().Update(context.TODO(), provider)`  I had to add the DeepCopy() func to make sure e all of the fields are copying properly.

Reply:
ah, yeah.
we hit this issue since we upgraded the controller-runtime once...
https://github.com/kubev2v/forklift/commit/71d5b9bb71600dcfc3917422b6a2a62434abc478